### PR TITLE
Use basic authentication instead of Header token

### PR DESCRIPTION
**Pull request by psyfear**

### CODE REVIEW

File: lib/promisepay_ex/client.ex [hunk 1]
```diff
@@ -7,25 +7,41 @@ defmodule PromisepayEx.Client do
...
     params
     |> URI.encode_query
     |> build_params(url)
-    |> send_request(token)
+    |> send_request(username, password)
+  end
+
+  defp build_basic_auth(nil, nil) do
+    []
   end
```
Comment (wnuqui) [inline lib/promisepay_ex/client.ex]: What about
```elixir
 def build_basic_auth(nil, nil), do: []
```
?

File: test/promisepay_ex_test.exs [hunk 2]
```diff
@@ -12,11 +12,12 @@ defmodule PromisepayExTest do
     )
 
     Config.filter_url_params(true)
-    Config.filter_request_headers("Authorization")
+    Config.filter_request_headers("basic_auth")
+    #Config.filter_sensitive_data("<PASSWORD>.+</PASSWORD>", "PLACEHOLDER")
```
Comment (wnuqui) [inline test/promisepay_ex_test.exs]: This can go away as we can always rely on commit history if ever we need to use this back again?

**Comment by psyfear** [comment]
thanks for the feedback @wnuqui

**Comment by wnuqui** [comment]
🚀

**Pull request merged**